src/Microsoft.CodeQuality.Analyzers/Core/Maintainability/RemoveUnusedLocals.Fixer.cs

```diff
@@ -1,27 +1,174 @@
+                            if (referencedSymbolNode != null)
+                            {
+                                var assignmentNode = GetNodeToRemoveOrReplace(referencedSymbolNode);
+                                if (assignmentNode != null)
+                                {
+                                    nodesToRemoveBuilder.Add(assignmentNode);
+                                }
+                            }
+                        }
+                    }
+                }
+
+                var declarationNode = editor.Generator.GetDeclaration(node);
+                nodesToRemoveBuilder.Add(node);
+
+                var nodesToRemove = nodesToRemoveBuilder.ToImmutable();
+                return nodesToRemove;
+            }
+
+            public async Task<Document> RemoveNodes(Document document, IEnumerable<SyntaxNode> nodes, CancellationToken cancellationToken)
+            {
+                DocumentEditor editor = await DocumentEditor.CreateAsync(document, cancellationToken).ConfigureAwait(false);
+                foreach (var node in nodes.Where(n => n != null).OrderByDescending(n => n.SpanStart))
+                {
+                    RemoveNode(editor, node);
+                }
+
+                return editor.GetChangedDocument();
+            }
+
+            public abstract void RemoveNode(DocumentEditor editor, SyntaxNode node);
+        }
+
+        private sealed class RemoveLocalAction : DocumentChangeAction
         {
-            // See https://github.com/dotnet/roslyn/blob/master/docs/analyzers/FixAllProvider.md for more information on Fix All Providers
-            return WellKnownFixAllProviders.BatchFixer;
+            public RemoveLocalAction(string title, Func<CancellationToken, Task<Document>> createChangedDocument)
+                : base(title, createChangedDocument) { }
+
+            public override string EquivalenceKey => null;
         }
 
-        public sealed override Task RegisterCodeFixesAsync(CodeFixContext context)
+        private sealed class RemoveLocalFixAllAction : CodeAction
         {
-            // Fixer not yet implemented.
-            return Task.CompletedTask;
+            private readonly Solution _solution;
+            private readonly Dictionary<Document, HashSet<SyntaxNode>> _nodesToRemoveMap;
+            private readonly NodesProvider _nodesProvider;
+
+            public RemoveLocalFixAllAction(Solution solution, Dictionary<Document, HashSet<SyntaxNode>> nodesToRemoveMap, NodesProvider nodesProvider)
+            {
+                _solution = solution;
+                _nodesToRemoveMap = nodesToRemoveMap;
+                _nodesProvider = nodesProvider;
+            }
+
+            public override string Title { get => MicrosoftMaintainabilityAnalyzersResources.RemoveUnusedLocalsMessage; }
+
+            protected override async Task<Solution> GetChangedSolutionAsync(CancellationToken cancellationToken)
```

Comment:
You do not need to override this method as your local (single) code fix cannot change more than one document.

Reply:
See http://source.roslyn.io/#Microsoft.CodeAnalysis.Workspaces/CodeActions/CodeAction.cs,161

Reply:
Sorry, my bad, I did not see this was inside `RemoveLocalFixAllAction`. Ignore this comment.